client/my-sites/stats/hooks/use-subscribers-query.tsx

```diff
@@ -1,53 +1,67 @@
-import { useQuery } from '@tanstack/react-query';
+import { useQuery, useQueries, UseQueryResult } from '@tanstack/react-query';
 import wpcom from 'calypso/lib/wp';
 
-interface SubscriberPayload {
+export interface SubscriberPayload {
 	date: string;
 	unit: string;
 	data: any[]; // TODO: add type
 	fields: string[];
 }
 
-export function querySubscribers(
+interface SubscribersData {
+	date: string;
+	unit: string;
+	data: {
+		[ key: string ]: string | number;
+	}[];
+}
+
+function querySubscribers(
 	siteId: number | null,
 	period: string,
 	quantity: number,
 	date?: string
-): Promise< any > {
-	// Create a date object from the passed date string or create a new date object if no date is provided.
+): Promise< SubscriberPayload > {
 	const periodStart = date ? new Date( date ) : new Date();
-
-	// Formatted date
 	const formattedDate = periodStart.toISOString().slice( 0, 10 );
 
-	const query: { unit: string; quantity: number; http_envelope: number; date: string } = {
+	const query = {
 		unit: period,
 		quantity,
 		http_envelope: 1,
 		date: formattedDate,
 	};
 
-	return wpcom.req.get(
-		{
-			method: 'GET',
-			apiNamespace: 'rest/v1.1',
-			path: `/sites/${ siteId }/stats/subscribers`,
-		},
-		query
-	);
+	return wpcom.req
+		.get(
+			{
+				method: 'GET',
+				apiNamespace: 'rest/v1.1',
+				path: `/sites/${ siteId }/stats/subscribers`,
+			},
+			query
+		)
+		.then( ( response ) => {
+			const selectedData = selectSubscribers( response );
+			return selectedData;
+		} );
 }
 
-export function selectSubscribers( payload: SubscriberPayload ) {
+function selectSubscribers( payload: SubscriberPayload ): SubscribersData {
 	if ( ! payload || ! payload.data ) {
-		return [];
+		// return default SubscribersData object if no payload data
+		return {
+			date: '',
```

Comment:
I don't think that passing empty strings instead of undefined values is the right approach. Could you return an empty object and use the optional chaining operator in the code consuming the data?